src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -0,0 +1,294 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockExpiredException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * MongoDbStore is a StoreInterface implementation using MongoDB as store engine.
+ *
+ * @author Joe Bennett <joe@assimtech.com>
+ */
+class MongoDbStore implements StoreInterface
+{
+    private $mongo;
+    private $options;
+    private $initialTtl;
+    private $collection;
+
+    /**
+     * @param \MongoDB\Client $mongo
+     * @param array           $options
+     *
+     * database: The name of the database [required]
+     * collection: The name of the collection [default: lock]
+     * resource_field: The field name for storing the lock id [default: _id] MUST be uniquely indexed if you chage it
+     * token_field: The field name for storing the lock token [default: token]
+     * acquired_field: The field name for storing the acquisition timestamp [default: acquired_at]
+     * expiry_field: The field name for storing the expiry-timestamp [default: expires_at].
+     *
+     * It is strongly recommended to put an index on the `expiry_field` for
+     * garbage-collection. Alternatively it's possible to automatically expire
+     * the locks in the database as described below:
+     *
+     * A TTL collections can be used on MongoDB 2.2+ to cleanup expired locks
```

Comment:
The current implementation does not purge expired locks. I'd rather use the verb MUST BE used,

Reply:
changed